packages/client-node/src/index.ts

```diff
@@ -13,43 +13,65 @@ import { Hash } from "@logion/node-api";
 import { Hash as Hasher } from 'fast-sha256';
 import FormData from "form-data";
 import fs from "fs";
+import { Readable } from "stream";
 
 
 export async function newLogionClient(env: Environment | EnvironmentString): Promise<LogionClient> {
     return await LogionClient.create(createConfig(env, () => new NodeAxiosFileUploader()))
 }
 
-export class NodeFile extends File {
+export abstract class NodeStreamFile extends File {
 
-    constructor(path: string, name: string, mimeType: MimeType) {
-        super(name, mimeType);
-        this.path = path;
-    }
-
-    private path: string;
-
-    async getHashAndSize(): Promise<HashAndSize> {
+    getHashAndSize(): Promise<HashAndSize> {
         const stream = this.getStream();
         return new Promise<HashAndSize>((resolve, reject) => {
             const digest = new Hasher();
             let size = 0n;
-            stream.on("data", data => {
+            stream.on("data", (data: Uint8Array) => {
                 size = size + BigInt(data.length);
                 digest.update(data);
             });
-            stream.on("end", () => resolve({
-                hash: Hash.fromDigest(digest),
-                size
-            }));
+            stream.on("end", () =>
+                resolve({
+                    hash: Hash.fromDigest(digest),
+                    size,
+                }),
+            );
             stream.on("error", reject);
         });
     }
 
+    abstract getStream(): NodeJS.ReadableStream;
+}
+
+export class NodeFile extends NodeStreamFile {
+
+    constructor(path: string, name: string, mimeType: MimeType) {
+        super(name, mimeType);
+        this.path = path;
+    }
+
+    private path: string;
+
     getStream(): NodeJS.ReadableStream {
         return fs.createReadStream(this.path);
     }
 }
 
+export class NodeBufferFile extends NodeStreamFile {
+
+    constructor(name: string, mimeType: MimeType, content: Buffer) {
```

Comment:
Would be nice to follow the same convention as in `NodeFile` and other adapters (see for instance [the constructor of an `ExpoFile`](https://github.com/logion-network/logion-api/blob/main/packages/client-expo/src/index.ts#L22)) and put the content as the first argument in the list.